### DISCUSSION TITLE
Sensors are not reinitializing on wake Hydro Series H150i Pro RGB

### DISCUSSION THREAD

**Discussion by mgbargeron** [EvanMulawski/FanControl.CorsairLink#205] [Q&A] [answered]
Hey, I've been having this issue after wake from sleep where the sensors are not reinitializing. I have a Hydro Series H150i Pro RGB. It works perfect before sleep then it craps the bed. I've tried this with 4.8 and 8.0. I tried 1.6 and all versions of 1.7beta. I have tried increasing the start time. I've got SiUSBXp.dll in the plugins directory. Corsair service is not running. (or even installed)

I'm really stuck here any help would be appreciated 

![Screenshot 2024-11-28 221021](https://github.com/user-attachments/assets/62cc9e63-610b-4e99-b3d1-28c856980fc6)
![Screenshot 2024-11-28 221149](https://github.com/user-attachments/assets/1c952993-fc0c-4ec8-a357-3f5b722123fc)
![Screenshot 2024-11-28 221213](https://github.com/user-attachments/assets/b01d0055-9c71-42ed-a004-208c5dac1539)

**Comment by EvanMulawski** [comment]
Hi @mgbargeron, I am working on a solution for this (see #89). As a workaround, you can use a program called [DevManView](https://www.nirsoft.net/utils/device_manager_view.html) to easily disable and re-enable (using its `disable_enable` command) the device. Configure a scheduled task in Windows to run that command [when resuming from sleep](https://sumtips.com/how-to/run-program-windows-wakes-up-sleep-hibernate/).

**Reply by mgbargeron** [reply]
this worked perfectly thank you so much! btw this plugin is great thanks for developing it. I'll post the script I used and the small demo of the setup for people who may not be as technical here in a second

**Comment by mgbargeron** [comment, marked as answer]
Warning: disabling the wrong device can cause problems with your machine. I'm not responsible if you mess up your machine. I'm only sharing what I did to make my machine work. 

1. Download [DevManView](https://www.nirsoft.net/utils/device_manager_view.html) 
- the download link is half way down the page 
- DevManView is 32-bit
- DevManView for x64 is 64-bit (most operating systems)
![Screenshot 2024-11-29 162017](https://github.com/user-attachments/assets/cef555d8-f362-4c16-b18c-35c47e5a46bd)

2. unzip the contents of the DevManView zip file and copy it to the root of your "C:" drive (it should be "devmanview" or "devmanview-64")

3. Run DevManView to find your Device Instance ID
![Screenshot 2024-11-29 165517](https://github.com/user-attachments/assets/bfb0af44-07de-478a-a961-e4245798cc84)


4. follow this guide provided above by Evan [Run Program After Wake From Sleep](https://sumtips.com/how-to/run-program-windows-wakes-up-sleep-hibernate/). When you get to the end of it, before you save it, see step 5-7 for the changes i made.  

5. In the General tab under the security tab check the box "Run with highest privledges"
![Screenshot 2024-11-29 164157](https://github.com/user-attachments/assets/e3f7920b-6148-4071-ac1e-9ab96d9b43fa)


6. In the Actions tab click New, Add your Start script, and click OK
- script format <path to .exe> /RunAsAdmin /disable_enable "<device instance id>"
- FOR EXAMPLE ONLY this is what mine looks like C:\devmanview-64\DevManView.exe /disable_enable "USB\VID_1B1C&PID_0C12\7289_2.0"
![Screenshot 2024-11-29 165306](https://github.com/user-attachments/assets/a88afcd9-cfa4-4b25-93f2-5a3ad9898eab)

7. click OK to save the script.

8. restart your computer and everything should be working.

**Reply by radwilly1** [reply]
Worked for me. Thanks so much!

**Reply by radwilly1** [reply]
> Worked for me. Thanks so much!

All of a sudden it's not working again :(
